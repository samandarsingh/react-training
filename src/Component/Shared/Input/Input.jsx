import React from "react";
import PropTypes from "prop-types";
import styles from "./Input.module.scss";
/**
 * this is input field component using in shared
 * @param {string} inputLabel 
 * @param {string} inputType
 * @param {string} inputName
 * @param {bool} isRequired
 * @param {string} inputValue
 * @param {func} handleChange
 * @param {number} inputMaxLenght
 * @returns element
 */ 
const Input = props => {
    const {
        inputLabel,
        inputType,
        inputName,
        isRequired,
        inputValue,        
        handleChange,
        inputMaxLenght
    } = props;
    return (
        <div>
            <label className={styles.labelClass}>{inputLabel}</label>
            <input 
                className={styles.inputClass}
                type={inputType}
                name={inputName}
                value={inputValue}
                required={isRequired}
                onChange={e => handleChange(e)}
                maxLength={inputMaxLenght}
                placeholder={`Please Enter ${inputLabel}`}            
            />
        </div>
    );
}
// default props
Input.defaultProps = {
inputLabel: "Todo",
inputType: "text",
inputName: "todo",
isRequired: true,
handleChange: () => {},
inputValue: "test",
inputMaxLenght: "30"
};
// proptypes validation
Input.propTypes = {
inputLabel: PropTypes.string.isRequired,
inputType: PropTypes.string.isRequired,
inputName: PropTypes.string.isRequired,
isRequired: PropTypes.bool,
handleChange: PropTypes.func,
inputValue: PropTypes.string,
inputMaxlength: PropTypes.string
};
export default Input;