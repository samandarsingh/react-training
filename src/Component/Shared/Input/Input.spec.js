import React from "react";
import { mount } from "enzyme";
import Input from "./Input";
describe(`Testing Input component`, () => {
    let wrapper = "";
    beforeAll(() => {
        const props = {
            inputLabel: "Todo",
            inputType: "text",
            inputName: "todo",
            isRequired: true,
            handleChange: () => {},
            inputValue: "test",
            inputMaxLenght: "30"
        }
        wrapper = mount(<Input {...props}/>);
    });
    it(`Input component render should be without crashing`, () => {
        expect(wrapper).toBeTruthy();
    });    
    it(`testing Input component getting props should be`, () => {        
        expect(
            wrapper
            .find(Input)
            .find(`label`)
            .hasClass(`labelClass`)
        ).toEqual(true);
        expect(
            wrapper
            .find(`input`)
            .props()
            .maxLength
        ).toEqual(`30`);
        
        expect(
             wrapper.find(`input`)
            .props()
            .required
        ).toEqual(true);
    });
    it(`testing Input component getting props onchange should be`, () => {        
        wrapper
        .find(`input`)
        .simulate(`change`,{ target : { value: `test`}});
        expect(
            wrapper
            .find(`input`)
            .props()    
            .value        
        ).toEqual(`test`);
    });
    it(`SnapShot testing`, () => {
        expect(wrapper).toMatchSnapshot();
    });
});