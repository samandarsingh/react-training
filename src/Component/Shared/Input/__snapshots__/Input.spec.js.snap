// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component SnapShot testing 1`] = `ReactWrapper {}`;
